lib/miq_expression.rb

```diff
@@ -1427,23 +1425,93 @@ def to_arel(exp, tz)
     end
   end
 
-  def extract_where_values(klass, scope)
-    relation = ActiveRecord::Relation.new(klass, klass.arel_table, klass.predicate_builder)
-    relation = relation.instance_eval(&scope)
+  def subquery_for_contains(field, subquery_limiter)
+    # base_class removes the default scopes. They are already in the main query and not needed in this subquery
+    main_model = field.model.base_class
 
-    begin
-      # This is basically ActiveRecord::Relation#to_sql, only using our
-      # custom visitor instance
-
-      connection = klass.connection
-      visitor    = WhereExtractionVisitor.new(connection)
-
-      arel  = relation.arel
-      binds = relation.bound_attributes
-      binds = binds.collect(&:value_for_database)
-      binds.map! { |value| connection.quote(value) }
-      collect = visitor.accept(arel.ast, Arel::Collectors::Bind.new)
-      collect.substitute_binds(binds).join
+    to_ref = field.reflections.first
+    join_keys = to_ref.join_keys
+    primary_attribute = field.model.arel_attribute(join_keys.foreign_key)
+    secondary_attribute = to_ref.klass.arel_attribute(join_keys.key)
+
+    # algorithm:
+    # 1. generate subquery as part of our main table since it has all the concepts we need.
```

Comment:
Maybe note that `.select(1)` is added here.  It kind of gets lots in the noise in the code below, and then a `    query_relation.projections.shift` happens, and it isn't clear where it came from.

That, or maybe add a:

```ruby
query_relation.projections.shift  # drop placeholder `SELECT 1` (added in step 1)
```